yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -163,6 +163,23 @@ private[spark] class Client(
     appContext.setQueue(args.amQueue)
     appContext.setAMContainerSpec(containerContext)
     appContext.setApplicationType("SPARK")
+    sparkConf.getOption(CONF_SPARK_YARN_APPLICATION_TAGS)
```

Comment:
Ah, I see. `setApplicationTags` takes a `Set[String]`, not a single string, so you're not calling it once per tag in the collection. My bad. Current code is fine.